src/DesignCompile/NetlistElaboration.cpp

```diff
@@ -1421,6 +1421,24 @@ bool NetlistElaboration::high_conn_(ModuleInstance* instance) {
       }
     }
   }
+  // Finally any unconnected ports with default value gets assigned the value
+  if (ports) {
+    for (auto p : *ports) {
+      if (p->High_conn() == nullptr) {
```

Comment:
You can simplify the super-nestedness here by just doing an early-out

```c++
for (auto p : *ports) {
  if (p->High_count() != nullptr) continue;
    ...
  if (found == allSignalss.end()) continue;
    ...
  if (NodeId defaultId != pair.second->getDefaultValue()) continue;
  ...
}
```

... and thus, from a very complicated, high-complexity code, it is very quick to parse for the human.